pkg/controller/core/localqueue_controller.go

```diff
@@ -336,12 +325,17 @@ func (r *LocalQueueReconciler) SetupWithManager(mgr ctrl.Manager, cfg *config.Co
 	queueCQHandler := qCQHandler{
 		client: r.client,
 	}
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&kueue.LocalQueue{}).
+	return builder.TypedControllerManagedBy[reconcile.Request](mgr).
+		Named("localqueue_controller").
+		WatchesRawSource(source.TypedKind(
+			mgr.GetCache(),
+			&kueue.LocalQueue{},
+			&handler.TypedEnqueueRequestForObject[*kueue.LocalQueue]{},
+			r,
+		)).
```

Comment:
This is an alternative as `For()`. If we want to use `TypedPredicate`, we need to use this `WatchesRawSource`.